app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt

```diff
@@ -137,19 +141,43 @@ open class SuggestionsViewModel(
     }
   }
 
-  fun showBottomSheet(comment: Comment) {
+  open fun showBottomSheet(comment: Comment) {
     viewModelScope.launch {
       _bottomSheetVisible.value = true
       _selectedComment.value = comment
     }
   }
 
-  private fun hideBottomSheet() {
+  open fun hideBottomSheet() {
     _bottomSheetVisible.value = false
   }
 
-  fun deleteComment() {
-    // Implement your delete logic here
+  open fun deleteComment(suggestion: Suggestion) {
+    // delete selected comment logic
+    val updatedSuggestion =
+        suggestion.copy(comments = suggestion.comments - _selectedComment.value!!)
+    viewModelScope.launch {
+      val wasUpdateSuccessful =
+          suggestionRepository?.updateSuggestionInTrip(tripId, updatedSuggestion)!!
+      if (wasUpdateSuccessful) {
+        loadSuggestion(tripId)
+      }
+    }
+    _selectedComment.value = null
+    hideBottomSheet()
+  }
+
+  open fun showDeleteDialog() {

```

Comment:
Great use of function to set values, makes the code more readable further down the line